autometa/taxonomy/gtdb.py

```diff
@@ -0,0 +1,368 @@
+        return self.dbdir
+
+    def search_genome_accessions(self, accessions: set) -> Dict[str, int]:
+        """
+        Search taxid.map file
+
+        Parameters
+        ----------
+        accessions : set
+            Set of subject sequence ids retrieved from diamond blastp search (sseqids)
+
+        Returns
+        -------
+        Dict[str, int]
+            Dictionary containing sseqids converted to taxids
+        """
+        sseqids_to_taxids = {}
+        # "rt" open the database in text mode instead of binary to be handled like a text file
+        fh = (
+            gzip.open(self.accession2taxid, "rt")
+            if is_gz_file(self.accession2taxid)
+            else open(self.accession2taxid)
+        )
+        # skip the header line
+        __ = fh.readline()
+        n_lines = file_length(self.accession2taxid)
+        converted_sseqid_count = 0
+        logger.debug(f"parsing {self.accession2taxid}...")
+        for line in fh:
+            acc_ver, taxid = line.strip().split("\t")
+            taxid = int(taxid)
+            if acc_ver in accessions:
+                sseqids_to_taxids[acc_ver] = taxid
+                converted_sseqid_count += 1
+        fh.close()
+        logger.debug(f"sseqids converted: {converted_sseqid_count:,}")
+        return sseqids_to_taxids
+
+    def parse_nodes(self) -> Dict[int, str]:
+        """
+        Parse the `nodes.dmp` database.
+        Note: This is performed when a new GTDB class instance is constructed
+
+        Returns
+        -------
+        dict
+            {child_taxid:{'parent':parent_taxid,'rank':rank}, ...}
+        """
+        fh = open(self.nodes_fpath)
+        __ = fh.readline()  # root line
+        nodes = {1: {"parent": 1, "rank": "root"}}
+        for line in tqdm(fh, desc="parsing nodes", leave=False):
+            child, parent, rank = line.split("\t|\t")[:3]
+            parent, child = [int(node) for node in [parent, child]]
+            rank = rank.lower()
+            nodes.update({child: {"parent": parent, "rank": rank}})
+        fh.close()
+        return nodes
+
+    def parse_names(self) -> Dict[int, str]:
+        """
+        Parses through names.dmp database and loads taxids with scientific names
+
+        Returns
+        -------
+        dict
+            {taxid:name, ...}
+        """
+        names = {}
+        fh = open(self.names_fpath)
+        for line in tqdm(fh, desc="parsing names", leave=False):
+            taxid, name, __, classification = line.strip("\t|\n").split("\t|\t")[:4]
+            taxid = int(taxid)
+            name = name.lower()
+            # Only add scientific name entries
+            if classification == "scientific name":
+                names.update({taxid: name})
+        fh.close()
+        return names
+
+    def parse_merged(self) -> Dict[int, int]:
+        """
+        Parse the merged.dmp database
+        Note: This is performed when a new NCBI class instance is constructed
```

Comment:
```suggestion
```

These docstrings (`parse_merged` and `parse_delnodes`) should be placed in the corresponding `TaxonomyDatabase` abstract methods, details specifically about GTDB may be placed here.